OpenDreamClient/Rendering/DreamViewOverlay.cs

```diff
@@ -103,84 +435,471 @@ sealed class DreamViewOverlay : Overlay {
+    public float AlphaToApply;
+    public Matrix3 TransformToApply;
+    public String RenderSource;
+    public String RenderTarget;
+    public List<RendererMetaData>? KeepTogetherGroup;
+    public int AppearanceFlags;
+    public int BlendMode;
+    public MouseOpacity MouseOpacity;
+
+    public RendererMetaData(){
+        Reset();
+    }
+
+    public void Reset(){
+        MainIcon = new DreamIcon();
+        Position = Vector2.Zero;
+        Plane = 0;
+        Layer = 0;
+        UID = EntityUid.Invalid;
+        ClickUID = EntityUid.Invalid;
+        IsScreen = false;
+        TieBreaker = 0;
+        ColorToApply = Color.White;
+        ColorMatrixToApply = null;
+        AlphaToApply = 1.0f;
+        TransformToApply = Matrix3.Identity;
+        RenderSource = "";
+        RenderTarget = "";
+        KeepTogetherGroup = null; //don't actually need to allocate this 90% of the time
+        AppearanceFlags = 0;
+        BlendMode = 0;
+        MouseOpacity = MouseOpacity.Transparent;
+    }
+
+    public int CompareTo(RendererMetaData other) {
+        int val = 0;
+
+        //Render target and source ordering is done first.
+        //Anything with a render target goes first
+        //Anything with a render source which points to a render target must come *after* that render_target
+        val = (this.RenderTarget.Length > 0).CompareTo(other.RenderTarget.Length > 0);
+        if (val != 0) {
+            return -val;
+        }
+        if(this.RenderSource.Length > 0)
+        {
+            if(this.RenderSource == other.RenderTarget)
+                return 1;
+        }
+
+        //We now return to your regularly scheduled sprite render order
+
+        //Plane
+        val =  this.Plane.CompareTo(other.Plane);
+        if (val != 0) {
+            return val;
+        }
+        val = (((int)this.AppearanceFlags & 128) == 128).CompareTo(((int)other.AppearanceFlags & 128) == 128); //appearance_flags & PLANE_MASTER
+        //PLANE_MASTER objects go first for any given plane
+        if (val != 0) {
+            return -val; //sign flip because we want 1 < -1
+        }
+        //subplane (ie, HUD vs not HUD)
+        val = this.IsScreen.CompareTo(other.IsScreen);
+        if (val != 0) {
+            return val;
+        }
+        //depending on world.map_format, either layer or physical position
+        //TODO
+        val = this.Layer.CompareTo(other.Layer);
+        if (val != 0) {
+            //special handling for EFFECTS_LAYER and BACKGROUND_LAYER
+            int effects_layer = (((int)this.Layer & 5000) == 5000).CompareTo(((int)other.Layer & 5000) == 5000);
+            if(effects_layer != 0)
+                return effects_layer;
+            int background_layer = (((int)this.Layer & 20000) == 20000).CompareTo(((int)other.Layer & 20000) == 20000);
+            if(background_layer != 0)
+                return -background_layer; //flipped because background_layer flag forces it to the back
```

Comment:
Rename `background_layer` to `backgroundLayer`
```suggestion
            int backgroundLayer = (((int)this.Layer & 20000) == 20000).CompareTo(((int)other.Layer & 20000) == 20000);
            if(backgroundLayer != 0)
                return -backgroundLayer; //flipped because background_layer flag forces it to the back
```